version: "3.8"
services:
  celery:
    container_name: celery
    restart: on-failure
    build:
      context: ../../my-setera-BACKEND
      dockerfile: Dockerfile
    environment:
      DB_HOST: db
      DB_USER: postgres
      DB_PASSWORD: postgres
      DB_NAME: postgres
      CELERY_BROKER_URL: "redis://redis:6379/0"
      CELERY_RESULT_BACKEND: "redis://redis:6379/0"
    volumes:
      - ../../my-setera-BACKEND/my_backend:/my_backend
    env_file:
      - ./.env
    command: celery  -A my_backend worker -l info
    depends_on:
      - redis
      - backend
    networks:
      setera-nw:

  celery-beat:
    container_name: celery_beat
    build:
      context: ../../my-setera-BACKEND
      dockerfile: Dockerfile
    environment:
      DB_HOST: db
      DB_USER: postgres
      DB_PASSWORD: postgres
      DB_NAME: postgres
      CELERY_BROKER_URL: "redis://redis:6379/0"
      CELERY_RESULT_BACKEND: "redis://redis:6379/0"
    volumes:
      - ../../my-setera-BACKEND/my_backend:/my_backend
    env_file:
      - ./.env
    command: celery -A my_backend beat -l info
    depends_on:
      - redis
      - backend
    networks:
      setera-nw:
